app_usb_aud_xk_316_mc/src/core/xua_conf.h

```diff
@@ -98,7 +98,7 @@
     #if (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN + 8*XUA_ADAT_TX_EN) <= NUM_USB_CHAN_OUT
         #define ADAT_TX_INDEX    (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN)
     #else
-        #define ADAT_TX_INDEX    (0)
+        #define ADAT_TX_INDEX    (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN)
```

Comment:
You might have to explain this one to me - ADAT_TX_INDEX can now be > NUM_USB_CHAN_OUT causing a trap?

Reply:
from what I understand, ADAT_TX_INDEX should always be I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN.
The #if (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN + 8*XUA_ADAT_TX_EN) <= NUM_USB_CHAN_OUT check assumes that there are 8 ADAT channels which is not true if we build configs such as min_freq=88, max_freq=96 and num_usb_chan_out=12. 
I can add a compile time check to ensure ADAT_TX_INDEX is less than NUM_USB_CHAN_OUT.